### PR TITLE
Test order of execution

### LINKED ISSUE #1343: Test more side-effects/order of execution
There should be a side-effects test-suite. In particular, trapping is a side-effect.

Code like this:

```
i32.const 0 i32.const 0 i32.div_s i32.const 1 global.set 0 local.set 0
```

breaks on an implementation that currently passes every other test. (the test should set the global to 0, call this code, and assert that the global is still 0.)

More importantly, this example is a fundamental core semantic. That's why it should be part of the spec test.

### PR DESCRIPTION
**Pull request by SoniEx2**
Fixes #1343.

### CODE REVIEW

File: test/core/order_of_execution.wast [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  (func $set_by_call (param i32)
+    (local.get 0) (global.set $a)
+  )
+
+  (func (export "trapping-div-trapping-call") (local i32)
+    (i32.const 0) (i32.const 0) (i32.div_s) (call $set_by_call (i32.const 3)) (local.set 0)
+  )
+)
+
+(assert_trap (invoke "div-by-zero") "integer divide by zero")
```
Comment (ngzhian) [inline test/core/order_of_execution.wast:26]: LGTM, maybe a comment here describing what we want these tests to catch? Along the lines of
"Traps have side effect and instructions follow them should not run, optimizations must preserve this side-effect"

**Review by rossberg — CHANGES_REQUESTED** [review]
Except for a few legacy exceptions, the test suite is structured around individual constructs (or groups thereof). Instead of introducing a new file with a random selection of things, can you put each test into the appropriate file for the affected construct?

**Comment by rossberg** [comment]
@SoniEx2, this PR is stale, do you still intend to land it?

**Comment by SoniEx2** [comment]
kinda haven't had the time to work on this anymore, sorry.

**Pull request closed without merging**